packages/smooth_app/lib/pages/product/common/product_refresher.dart

```diff
@@ -112,6 +112,16 @@ class ProductRefresher {
   }) async =>
       _fetchAndRefreshList(localDatabase, barcodes, productType);
 
+  /// Fetches the products from the server, refreshes the local database.
+  /// And returns a list of barcodes for which a product was found.
+  /// Silent version.
+  Future<List<String>?> silentFetchAndRefreshListWithFeedback({
```

Comment:
Then what's the purpose of this method as it replays exactly another method?
For the record the "WithFeedback" part of the naming is very misleading.

Reply:
I wasn't sure if I could make _fetchAndRefreshList public. What would you recommend here?

Do you know why this method was private in the first place? Should we add checks before calling it?

Reply:
By default methods should be private. In that case (`silentFetchAndRefreshList` and `_fetchAndRefreshList`) I guess we used to have also something like `fetchAndRefreshList` that displays a dialog while refreshing a list. That doesn't seem to be the case anymore. 

In the current case, I would recommend
* a single method
* instead of the already existing `silentFetchAndRefreshList` and `fetchAndRefreshList`, and your new `silentFetchAndRefreshListWithFeedback`
* from the existing `Future<void> silentFetchAndRefreshList`
* called `Future<List<String>> silentFetchAndRefreshList`